main/src/ca/uwaterloo/flix/language/ast/JvmAst.scala

```diff
@@ -25,6 +25,7 @@ object JvmAst {
                   enums: Map[Symbol.EnumSym, Enum],
                   structs: Map[Symbol.StructSym, Struct],
                   effects: Map[Symbol.EffSym, Effect],
+                  erasedFunctionTypes: Set[SimpleType.Arrow],
                   types: Set[SimpleType],
```

Comment:
The remaining classes that are generated based on types are:
- Tuple
- Lazy
- Records

So if we add those in the same way that `erasedFunctionTypes` is added here, `types` can be deleted along with its functions `allTypes`, `nestedTypesOf`, `extractFunctionType`, `typesOfEffect`. The three remaining classes should only take 20-50 lines to add now that `visitType` is here.